docs/table_design/list_partitioning.md

```diff
@@ -0,0 +1,110 @@
+# List partitioning
+
+Since v3.1, StarRocks supports List partitioning. Data is partitioned based on a predefined value list  for each partition, which can accelerate queries and facilitate management according to enumerated values.
+
+## Introduction
+
+You need to explicitly specify the column values list in each partition. These values do not need to be continuous, unlike the continuous time or numeric range required in Range Partitioning. During data loading, StarRocks will store the data in the corresponding partition based on the mapping between the data's partition column values and the predefined column values for each partition.
+
+![list_partitioning](../assets/list_partitioning.png)
+
+List partitioning is suitable for storing data whose columns contain a small number of emurated values, and you often query and manage data based on these emunrated values. For example, columns represent geographical locations, states, and categories. Each value in a column represents an independent category. By partitioning data based on emurated column values, you can improve query performance and facilitate data management.
+
+**List Partitioning is particularly useful for scenarios where a partition needs to include multiple values for each partition column**. For example, if a table includes the "City" column representing the city to which an individual belongs, and you often query and manage data by states and cities. At table creation, you can use the `City`  column as the partition column for List Partitioning and specify that data of various cities within the same state are placed in one partition, for example `PARTITION pCalifornia VALUES IN ("Los Angeles", "San Francisco", "San Diego")`. This method can accelerate queries based on states and cities while facilitating data management.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/5ed1992b-d9ab-4673-a8ea-247761aa4cb8)
column names should be wrapped in code format, not quotation marks